dataflow/src/main/java/org/checkerframework/dataflow/analysis/AnalysisResult.java

```diff
@@ -443,4 +445,124 @@ public S getStoreAfter(Block block) {
         return transferInput.analysis.runAnalysisFor(
                 node, before, transferInput, nodeValues, analysisCaches);
     }
+
+    /**
+     * Returns a string representation of this.
+     *
+     * @return a string representation of this
+     */
+    public String repr() {
+        StringJoiner result =
+                new StringJoiner(
+                        String.format("%n  "),
+                        String.format("AnalysisResult{%n  "),
+                        String.format("%n}"));
+        result.add("nodeValues = " + nodeValuesRepr(nodeValues));
+        result.add("treeLookup = " + treeLookupRepr(treeLookup));
+        result.add("unaryAssignNodeLookup = " + unaryAssignNodeLookup);
+        result.add("finalLocalValues = " + finalLocalValues);
+        result.add("stores = " + stores);
+        result.add("analysisCaches = " + analysisCaches);
+        return result.toString();
+    }
+
+    /**
+     * Return a printed representation of a map with the same type as the {@code nodeValues} field.
+     *
+     * @param <V> the type of values in the map
+     * @param nodeValues a map to format
+     * @return a printed representation of the given map
+     */
+    public static <V> String nodeValuesRepr(Map<Node, V> nodeValues) {
+        if (nodeValues.isEmpty()) {
+            return "{}";
+        }
+        StringJoiner result = new StringJoiner(String.format("%n    "));
+        result.add("{");
+        for (Map.Entry<Node, V> entry : nodeValues.entrySet()) {
+            Node key = entry.getKey();
+            result.add(String.format("%s => %s", nodeRepr(key), entry.getValue()));
+        }
+        result.add("}");
+        return result.toString();
+    }
+
+    /**
+     * Return a printed representation of a node.
+     *
+     * @param n a node to format
+     * @return a printed representation of the given node
+     */
+    public static String nodeRepr(Node n) {
```

Comment:
Any reason for not making this `Node.toString`?

Reply:
Same reason.